panic: unwait called without all parents blocked

Repro is with Zed commit 7de3ffa.

This bug appears to be a side effect of some of the Zed appended by the app in https://github.com/brimdata/brim/pull/2355 to help with pagination.

To repro, start a Zed service:

```
$ zed -version
Version: v1.4.0-16-g7de3ffa1

$ rm -rf lake

$ zed -lake lake serve
{"level":"info","ts":1673975387.540253,"logger":"core","msg":"Started"}
...
```

Create a pool, load the HTTP [zed-sample-data](https://github.com/brimdata/zed-sample-data), and execute the query shown.

```
$ zed create http
pool created: http 2KSkbbmNVx4r1eMBgcWFmTx1np0

$ zed -use http load http.log.gz 
(1/1) 5.52MB/5.52MB 5.52MB/s 100.00%
2KSkc0Li8a4zMx3NkHgXcxzAObb committed

$ zed query 'from "http" | _path=="http" | method !=null | count() by ts:=bucket(ts,20ms),method | { i: count(), v: this} | i > 0 | head 2000'
{i:1(uint64),v:{ts:2018-03-24T17:36:30Z,method:"POST",count:20(uint64)}}
{i:2(uint64),v:{ts:2018-03-24T17:36:29.98Z,method:"GET",count:1(uint64)}}
...
```

At this point the Zed service crashes with the following panic:

```
panic: unwait called without all parents blocked

goroutine 110 [running]:
github.com/brimdata/zed/runtime/op/combine.(*Proc).unwait(0x1?)
	/Users/phil/work/zed/runtime/op/combine/combine.go:89 +0x13d
github.com/brimdata/zed/runtime/op/combine.(*Proc).propagateDone(0xc0005444b0)
	/Users/phil/work/zed/runtime/op/combine/combine.go:154 +0x14e
github.com/brimdata/zed/runtime/op/combine.(*Proc).Pull(0xc0005444b0, 0x44?)
	/Users/phil/work/zed/runtime/op/combine/combine.go:44 +0x7f
github.com/brimdata/zed/runtime/op/groupby.(*Proc).sendResult(0xc0005445a0, {0x2259060, 0xc002237f50}, {0x0, 0x0})
	/Users/phil/work/zed/runtime/op/groupby/groupby.go:265 +0x162
github.com/brimdata/zed/runtime/op/groupby.(*Proc).run(0xc0005445a0)
	/Users/phil/work/zed/runtime/op/groupby/groupby.go:245 +0x42f
created by github.com/brimdata/zed/runtime/op/groupby.(*Proc).Pull.func1
	/Users/phil/work/zed/runtime/op/groupby/groupby.go:168 +0x6d
```

If I remove the `head 2000` from the end of the Zed pipeline, though, it doesn't crash.

--- a/runtime/op/combine/combine.go
+++ b/runtime/op/combine/combine.go
@@ -7,6 +7,7 @@ import (
 
 	"github.com/brimdata/zed/runtime/op"
 	"github.com/brimdata/zed/zbuf"
+	"golang.org/x/sync/errgroup"
 )
 
 type Proc struct {
@@ -109,18 +110,16 @@ func (p *Proc) block(parent *puller) {
 
 func (p *Proc) propagateDone() error {
 	var mu sync.Mutex
-	var wg sync.WaitGroup
+	var group errgroup.Group
 	for _, parent := range p.parents {
 		if parent.blocked {
 			continue
 		}
 		parent := parent
-		wg.Add(1)
 		// We use a goroutine here because sending to parents[i].doneCh
 		// can block until we've sent to parents[i+1].doneCh, as with
 		// "fork (=> count() => pass) | head".
-		go func() {
-			defer wg.Done()
+		group.Go(func() error {
 		again:
 			select {
 			case <-p.queue:
@@ -134,11 +133,15 @@ func (p *Proc) propagateDone() error {
 				mu.Lock()
 				p.block(parent)
 				mu.Unlock()
+				return nil
 			case <-p.ctx.Done():
+				return p.ctx.Err()
 			}
-		}()
+		})
 	}
-	wg.Wait()
+	if err := group.Wait(); err != nil {
+		return err
+	}
 	// Make sure all the dones that canceled pending queue entries
 	// are clear.  Otherwise, this will block the queue on the next
 	// platoon.

